src/lib.rs

```diff
@@ -1236,6 +1236,58 @@ pub trait Itertools : Iterator {
     {
         self.sorted_by(cmp)
     }
+
+    /// Return the minimum and maximum element of an iterator.
+    ///
+    /// For the minimum, the first minimal element is returned.  For the maximum,
+    /// the last maximal element wins.  This matches the behavior of the standard
+    /// `Iterator::min()` and `Iterator::max()` methods.
+    ///
+    /// This requires `clone()`-able elements since the returned minimum and
+    /// maximum might actually be the same item.
```

Comment:
The old min max in libstd handled this by using a None / One / Two kind of enum. It's a tradeoff, but I think the best is to support non-Clone elements.

A good ergonomic solution may be a method on the result enum that converts the result to a pair of values, defined only if the result is clonable?

Old code: https://github.com/rust-lang/rust/blob/d07d465cf60033e35eba16b9e431471d54c712f4/src/libcore/iter.rs#L809-L891